### ISSUE TITLE
blank lines between list item child nodes

### ISSUE THREAD

**Issue by ssikuro** [vsch/flexmark-java#341]
I have this Markdown document:
```
1. List item 1:
   * List item 1 sub list 1 item 1

   List item 1 para 1
```

I am using the FlexMark-java parser configured with options:
```
            .set(Parser.EXTENSIONS, Arrays.asList(StrikethroughSubscriptExtension.create(),
                                                  TablesExtension.create(),
                                                  YamlFrontMatterExtension.create()
                                                  ))
            .set(Parser.HEADING_NO_ATX_SPACE, true)
            .set(Parser.BLANK_LINES_IN_AST, true)
```

This parser parses the Markdown document as:
```
Document[0, 75]
 OrderedList[0, 74] isLoose delimiter:'.'
  OrderedListItem[0, 74] open:[0, 2, "1."] isLoose hadBlankLine
   Paragraph[3, 16]
    Text[3, 15] chars:[3, 15, "List  … em 1:"]
   BulletList[19, 51] isTight
    BulletListItem[19, 51] open:[19, 20, "*"] isTight hadBlankLineAfter
     Paragraph[21, 51] isTrailingBlankLine
      Text[21, 50] chars:[21, 50, "List  … tem 1"]
   Paragraph[55, 74] isTrailingBlankLine
    Text[55, 73] chars:[55, 73, "List  … ara 1"]
 BlankLine[74, 75]
```

Why isn't there a BlankLine node between the BulletListItem and the third and the last Paragraph node (representing "List item 1 para 1") as a sibling node?

Related to this, I have these questions:
* What does hadBlankLine of a LiteItem mean?
* What does hadBlankLineAfter of a ListItem mean?
* What does isTralingBlankLine of a Paragraph mean?

I'm using FlexMark-Java 0.32.24.

**Comment by vsch** [comment]
@ssikuro, the issue was fixed in a later release along with other blank line issues in the AST, major fix in 0.35.2 but also some fixes in 0.40.2. However, there were more changes to packages in 0.40.0 requiring more migration work so you should probably update at least to 0.35.8.

```````````````````````````````` example(Issue 341: 1) options(keep-blank-lines)
1. List item 1:
   * List item 1 sub list 1 item 1

   List item 1 para 1
.
<ol>
  <li>
    <p>List item 1:</p>
    <ul>
      <li>List item 1 sub list 1 item 1</li>
    </ul>
    <p>List item 1 para 1</p>
  </li>
</ol>
.
Document[0, 73]
  OrderedList[0, 73] isLoose delimiter:'.'
    OrderedListItem[0, 73] open:[0, 2, "1."] isLoose hadBlankLine
      Paragraph[3, 16]
        Text[3, 15] chars:[3, 15, "List  … em 1:"]
      BulletList[19, 51] isTight
        BulletListItem[19, 51] open:[19, 20, "*"] isTight hadBlankLineAfter
          Paragraph[21, 51] isTrailingBlankLine
            Text[21, 50] chars:[21, 50, "List  … tem 1"]
      BlankLine[51, 52]
      Paragraph[55, 73]
        Text[55, 73] chars:[55, 73, "List  … ara 1"]
````````````````````````````````

**Comment by vsch** [comment]
@ssikuro,

* `hadBlankLine` means that the list item had a blank line somewhere in its content, not necessarily a trailing blank line.
* `hadBlankLineAfter` means that the item is followed by a blank line.
* `isTrailingBlankLine` in paragraph means the paragraph was followed by a blank line.

**Comment by ssikuro** [comment]
Thank you for the quick response.
I wanted to try 0.35.8 as you suggested but I could not find that exact version in public maven repositories I've tried. Instead, I'm using 0.35.10. This one fixed the issue! But probably because of a minute change in the behavior, a few test cases I have are failing. Hopefully these are something I can deal with. (One failure suggests the ! prefix of external link references is doubled. Is this something you are familiar with?)

**Comment by vsch** [comment]
@ssikuro, just realized that 0.35.8 was published to maven as 0.36.8, but you are right 0.35.10 is the latest release in that branch.

Can you give me an example of what you mean by:

> One failure suggests the ! prefix of external link references is doubled.

I am not aware of this issue.

**Comment by ssikuro** [comment]
It looks like ImageRef.getTextOpeningMarker().toString() was returning "[" in ver. 0.32.24, and I had a piece of code to add "!". This changed somewhere before 0.35.10, and it now returns "![". So my code was replacing "![" with "!![". I changed my code and it's now working fine.
